File: app/src/main/java/com/example/udemy_thecompleteandroid14developercourse_build100apps/RecyclerViewTheGroceryApp/GroceryAdapter.java
package com.example.udemy_thecompleteandroid14developercourse_build100apps.RecyclerViewTheGroceryApp;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.udemy_thecompleteandroid14developercourse_build100apps.R;

import java.util.List;

public class GroceryAdapter extends RecyclerView.Adapter<GroceryAdapter.ViewHolder> {
    private final List<Grocery> groceryList;

    public GroceryItemClickListener clickListener;
    public void setClickListener(GroceryItemClickListener clickListener){
        this.clickListener = clickListener;
    }

    public GroceryAdapter(List<Grocery> groceryList){
        this.groceryList = groceryList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout_for_rv, parent, false);

        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Grocery grocery = groceryList.get(position);
        holder.title.setText(grocery.getGroceryName());
        holder.description.setText(grocery.getGroceryDescription());
        holder.groceryImage.setImageResource(grocery.getGroceryImage());
    }

    @Override
    public int getItemCount() {
        return groceryList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        ImageView groceryImage;
        TextView title;
        TextView description;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            groceryImage = itemView.findViewById(R.id.groceryImage);
            title = itemView.findViewById(R.id.groceryName);
            description = itemView.findViewById(R.id.groceryDescription);

            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            if (clickListener != null){
                clickListener.onClick(v, getAdapterPosition());
            }
        }
    }
}

File: app/src/main/java/com/example/udemy_thecompleteandroid14developercourse_build100apps/KotlinGroceryApp/adapter/GroceryAdapter.kt
package com.example.udemy_thecompleteandroid14developercourse_build100apps.KotlinGroceryApp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.example.udemy_thecompleteandroid14developercourse_build100apps.KotlinGroceryApp.model.Grocery
import com.example.udemy_thecompleteandroid14developercourse_build100apps.R

class GroceryAdapter(val groceryList: ArrayList<Grocery>):RecyclerView.Adapter<GroceryAdapter.GroceryViewHolder>() {
    
    inner class GroceryViewHolder(itemView: View):RecyclerView.ViewHolder(itemView){
        var groceryImage:ImageView = itemView.findViewById(R.id.groceryImageView)
        var groceryName:TextView = itemView.findViewById(R.id.groceryNameTextView)
        var groceryDesc:TextView = itemView.findViewById(R.id.groceryDescTextView)
        
        init {
            itemView.setOnClickListener{
                Toast.makeText(itemView.context, "You choose: ${groceryList[adapterPosition].name}", Toast.LENGTH_SHORT).show()
            }
        }
        
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GroceryViewHolder {
        val view:View = LayoutInflater.from(parent.context).inflate(R.layout.grocery_item_layout, parent, false)
        return GroceryViewHolder(view)
    }

    override fun getItemCount(): Int {
        return groceryList.size
    }

    override fun onBindViewHolder(holder: GroceryViewHolder, position: Int) {
        val currentGroceryItem:Grocery = groceryList[position]
        holder.groceryName.text = currentGroceryItem.name
        holder.groceryDesc.text = currentGroceryItem.desc
        holder.groceryImage.setImageResource(currentGroceryItem.image)
        
    }
}